sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientAsyncTest.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void listRepositories(HttpClient httpClient) {
+        client = getContainerRegistryAsyncClient(httpClient);
+        StepVerifier.create(client.listRepositories().byPage())
+            .assertNext(res -> validateListResponse(res))
+            .verifyComplete();
+    }
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void listRepositoriesWithNullOptions(HttpClient httpClient) {
+        client = getContainerRegistryAsyncClient(httpClient);
+        assertThrows(NullPointerException.class, () -> client.listRepositories(null));
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void listRepositoriesWithOptions(HttpClient httpClient) {
+        client = getContainerRegistryAsyncClient(httpClient);
+        ListRepositoriesOptions options = new ListRepositoriesOptions();
+        StepVerifier.create(client.listRepositories(options).byPage())
+            .assertNext(res -> validateListResponse(res))
+            .verifyComplete();
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void listRepositoriesWithPageSize(HttpClient httpClient) {
+        client = getContainerRegistryAsyncClient(httpClient);
+        ListRepositoriesOptions options = new ListRepositoriesOptions().setMaxPageSize(DEFAULTPAGESIZE);
+        StepVerifier.create(client.listRepositories(options).byPage())
+            .assertNext(res -> {
+                List<String> repositories = res.getValue();
+                assertEquals(1, repositories.size());
+                assertEquals(repositories.get(0), "dockercloud/hello-world");
+            })
+            .assertNext(res -> {
+                List<String> repositories = res.getValue();
+                assertEquals(1, repositories.size());
+                assertEquals(repositories.get(0), "library/hello-seattle");
+            })
+            .assertNext(res -> {
+                List<String> repositories = res.getValue();
+                assertEquals(1, repositories.size());
+                assertEquals(repositories.get(0), "library/hello-world");
+            })
+            .verifyComplete();
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void deleteRepositoriesWithNull(HttpClient httpClient) {
+        client = getContainerRegistryAsyncClient(httpClient);
+        StepVerifier.create(client.deleteRepository(null))
+            .expectError(NullPointerException.class)
+            .verify();
+    }
+
+
+    // What is the prescribed way of running delete test scenarios.
+/*    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void deleteRepositories(HttpClient httpClient) {
```

Comment:
The SDK does not support an API to add so I either need to run this test in the end\predictable order  or run the test-resources-post again as a before step, however I could not figure out a way to do either.